newfile.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type Book struct {
+	ID     string  `json:"id"`
+	ISBN   string  `json:"isbn"`
+	Title  string  `json:"title"`
+	Author *Author `json:"author"`
+}
+
+type Author struct {
+	Firstname string `json:"firstname"`
+	Lastname  string `json:"lastname"`
+}
+
+var books []Book
+
+func getBooks(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	w.Header().Set("Date", "Wed, 16 Aug 2022 12:15:38 GMT")
+	json.NewEncoder(w).Encode(books)
+}
+
+func getBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range books {
+		if item.ID == params["id"] {
+			json.NewEncoder(w).Encode(item)
+			return
+		}
+	}
+	// json.NewEncoder(w).Encode(&Book{})
+	fmt.Fprintf(w, "No Data available")
+}
+
+func createBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	var book Book
+	_ = json.NewDecoder(r.Body).Decode(&book)
+	book.ID = strconv.Itoa(rand.Intn((10000000)))
+	books = append(books, book)
+	json.NewEncoder(w).Encode(book)
+	fmt.Fprintf(w, "New Entry created and added to the database")
+}
+
+func updateBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range books {
+		if item.ID == params["id"] {
+			books = append(books[:index], books[index+1:]...)
+			var book Book
+			_ = json.NewDecoder(r.Body).Decode(&book)
+			books = append(books, book)
+			json.NewEncoder(w).Encode(book)
+			fmt.Fprintf(w, "Data of the current entry has been updated")
+			return
+		}
+	}
+	json.NewEncoder(w).Encode(books)
+}
+func deleteBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range books {
+		if item.ID == params["id"] {
+			books = append(books[:index], books[index+1:]...)
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(books)
+	fmt.Fprintf(w, "The current entry hass been deleted")
+}
+func main() {
+	r := mux.NewRouter()
+
+	books = append(books, Book{ID: "1", ISBN: "7741852", Title: "Char", Author: &Author{Firstname: "Sachin", Lastname: "Shinde"}},
+		Book{ID: "2", ISBN: "9963852", Title: "Pahunchaya", Author: &Author{Firstname: "Virat", Lastname: "Patil"}})
+
+	r.HandleFunc("/api/books", getBooks).Methods("GET")
+	r.HandleFunc("/api/books/{id}", getBook).Methods("GET")
+	r.HandleFunc("/api/books", createBook).Methods("POST")
+	r.HandleFunc("/api/books/{id}", updateBook).Methods("PUT")
+	r.HandleFunc("/api/books/{id}", deleteBook).Methods("DELETE")
```

Comment:
Instead of creating string like `GET`, `POST`, `PUT`, `DELETE` you can use existing constant from [http](https://pkg.go.dev/net/http#pkg-constants) package like `http.MethodGet`, `http.MethodPost`, `http.MethodPut`, etc.

Reply:
Ok